src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -341,32 +345,57 @@ public static FormulaValue Text(FormattingInfo formatInfo, IRContext irContext,
                 return CommonErrors.GenericInvalidArgument(irContext, string.Format(CultureInfo.InvariantCulture, customErrorMessage, formatSize));
             }
 
-            if (formatString != null && !TextFormatUtils.IsValidFormatArg(formatString, out bool hasDateTimeFmt, out bool hasNumberFmt))
+            if (formatString != null && !TextFormatUtils.IsValidFormatArg(formatString, out hasDateTimeFmt, out hasNumberFmt, out startIdx, out endIdx))
             {
                 var customErrorMessage = StringResources.Get(TexlStrings.ErrIncorrectFormat_Func, culture.Name);
                 return CommonErrors.GenericInvalidArgument(irContext, string.Format(CultureInfo.InvariantCulture, customErrorMessage, "Text"));
             }
 
-            var isText = TryText(formatInfo, irContext, args[0], formatString, out StringValue result);
+            var isText = TryText(formatInfo, irContext, args[0], hasDateTimeFmt, hasNumberFmt, formatString, startIdx, endIdx, out StringValue result);
 
             return isText ? result : CommonErrors.GenericInvalidArgument(irContext, StringResources.Get(TexlStrings.ErrTextInvalidFormat, culture.Name));
         }
 
-        public static bool TryText(FormattingInfo formatInfo, IRContext irContext, FormulaValue value, string formatString, out StringValue result)
+        public static bool TryText(FormattingInfo formatInfo, IRContext irContext, FormulaValue value, bool hasDateTimeFmt, bool hasNumberFmt, string formatString, int startIdx, int endIdx, out StringValue result)
```

Comment:
Are there any values of CCC and XXX where these are not the same?

>> Text(1234.5678, "[$-CCCCCCCC]XXXXXXXX")
  
>> Text(1234.5678, "XXXXXXXX", "CCCCCCCC")

Reply:
If yes, then we should definitely have a test case for that. 
If no, then we should normalize them to a single (formatString, CultureInfo) in this function.

Reply:
updated to handle single format